app/controllers/replies_controller.rb

```diff
@@ -297,19 +306,153 @@ def require_edit_permission
     redirect_to post_path(@reply.post)
   end
 
-  def preview(written)
-    @written = written
-    @post = @written.post
+  def get_multi_replies
+    # Get the multi replies stored in the page JSON
+    @multi_replies_params = JSON.parse(params.fetch(:multi_replies_json, "[]")).map do |reply_json|
+      permitted_params(ActionController::Parameters.new({ reply: reply_json }), [:id])
+    end
+    @multi_replies = @multi_replies_params.map do |reply_params|
+      Reply.new(reply_params).tap { |r| r.user = current_user }
+    end
+  end
```

Comment:
...it feels like there should be a more elegant way to do this, is there not?

Reply:
Not one I know, @Throne3d is the one who suggested that one iirc